PublicFolders/SourceSideValidations/SourceSideValidations.ps1

```diff
@@ -117,29 +117,48 @@ try {
 
     Write-Progress @progressParams -Status "Step 1 of 6"
 
-    $folderData = Get-FolderData -StartFresh $StartFresh -SlowTraversal $SlowTraversal
-
-    if ($folderData.IpmSubtree.Count -lt 1) {
-        return
-    }
+    $mailboxToServerMap = @{}
 
-    $script:anyDatabaseDown = $false
-    Get-Mailbox -PublicFolder | ForEach-Object {
+    # Validate that all PF mailboxes are available
+    $anyPFMailboxUnavailable = $false
+    $pfMailboxes = Get-Mailbox -PublicFolder
+    foreach ($mailbox in $pfMailboxes) {
         try {
-            $db = Get-MailboxDatabase $_.Database -Status
+            $db = Get-MailboxDatabase $mailbox.Database -Status
             if ($db.Mounted) {
-                $folderData.MailboxToServerMap[$_.DisplayName] = $db.Server
+                $mailboxToServerMap[$mailbox.DisplayName] = $db.Server
             } else {
-                Write-Error "Database $db is not mounted. This database holds PF mailbox $_ and must be mounted."
-                $script:anyDatabaseDown = $true
+                Write-Warning "Database $db is not mounted. This database holds PF mailbox $mailbox and must be mounted."
+                $anyPFMailboxUnavailable = $true
             }
         } catch {
             Write-Error $_
-            $script:anyDatabaseDown = $true
+            $anyPFMailboxUnavailable = $true
         }
     }
 
-    if ($script:anyDatabaseDown) {
+    $folderData = Get-FolderData -StartFresh $StartFresh -SlowTraversal $SlowTraversal
+
+    if ($folderData.IpmSubtree.Count -lt 1) {
+        return
+    }
+
+    $folderData.MailboxToServerMap = $mailboxToServerMap
+
+    # Validate that all content mailboxes exist
+    $ipmSubtreeByMailboxGuid = $folderData.IpmSubtree | Group-Object ContentMailboxGuid
+    foreach ($group in $ipmSubtreeByMailboxGuid) {
+        if ([string]::IsNullOrEmpty($group.Name)) {
+            Write-Warning "$($group.Count) folders have no ContentMailboxGuid. "
```

Comment:
Is this a point to attempt to get the mailbox information after this point if the name is in fact null or empty?

Just trying to understand the reason for this `if` statement as the result will come back with a `$null` value and provide you with the same information on line 156.

Reply:
Good point. I added that when I was trying to produce a null value, but I was never able to actually do that. I think we can just let it fall through to the existing logic.